templates/allocation/allocation_list.html

```diff
@@ -0,0 +1,38 @@
+{% extends "base.html" %}
+
+
+{% block content %}
+    <div class="p-2 w-50 mx-auto">
+        <h2>Allocation Settings</h2>
+        <a class="mb-4" href="/admin/consvc_shepherd/allocationsetting/add/">Add New Position</a>
+    {% if not allocation_settings %}
+    <h5>No allocations set.</h5>
+    {% endif %}
+        {% for a_s in allocation_settings %}
+            <div class="card p-2 mt-4 ">
+                <div class="px-2 d-flex justify-content-between align-items-center">
+                    <h4> Position {{ a_s.position }}</h4>
+                    <a href="/admin/consvc_shepherd/allocationsetting/{{ a_s.id }}/change/">edit</a>
+                </div>
+                <table class="table ">
+                    <thead class="text-secondary">
+                    <th>Partner</th>
+                    <th>Percentage (%)</th>
+                    </thead>
+                    <tbody>
+                    {% for p_a in a_s.partner_allocations.all %}
+                        <tr>
+                            <td>
+                                {{ p_a.partner }}
+                            </td>
+                            <td>
+                                {{ p_a.percentage }}
+                            </td>
+                        </tr>
+                    {% endfor %}
+                    </tbody>
+                </table>
+            </div>
+        {% endfor %}
```

Comment:
Would this be a good place to use `for...empty`? 🤔

```suggestion
    {% for a_s in allocation_settings %}
        <div class="card p-2 mt-4 ">
            <div class="px-2 d-flex justify-content-between align-items-center">
                <h4> Position {{ a_s.position }}</h4>
                <a href="/admin/consvc_shepherd/allocationsetting/{{ a_s.id }}/change/">edit</a>
            </div>
            <table class="table ">
                <thead class="text-secondary">
                    <th>Partner</th>
                    <th>Percentage (%)</th>
                </thead>
                <tbody>
                {% for p_a in a_s.partner_allocations.all %}
                    <tr>
                        <td>
                            {{ p_a.partner }}
                        </td>
                        <td>
                            {{ p_a.percentage }}
                        </td>
                    </tr>
                {% endfor %}
                </tbody>
            </table>
        </div>
    {% empty %}
        <h5>No allocations set.</h5>
    {% endfor %}
```

Reply:
oh this is cool! i didn't know you could do this

Reply:
Whoa, me neither, that's so cool and efficient! Awesome suggestion @linabutler .

Reply:
[Here's where I learned about it!](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#for-empty)